### PR TITLE
Add check for shunt compensator on section count

### PR DESCRIPTION
**Pull request by lilian-houdelet**

### CODE REVIEW

**Review by ayolab — APPROVED** [review]

File: src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorCreation.java [hunk 1]
```diff
@@ -34,6 +34,14 @@ public void check(Network network) throws NetworkModificationException {
         if (network.getShuntCompensator(modificationInfos.getEquipmentId()) != null) {
             throw new NetworkModificationException(SHUNT_COMPENSATOR_ALREADY_EXISTS, modificationInfos.getEquipmentId());
         }
+
+        if (modificationInfos.getMaximumSectionCount() < 1) {
+            throw new NetworkModificationException(CREATE_SHUNT_COMPENSATOR_ERROR, "Maximum section count should be greater or equal to 1");
+        }
+
+        if (modificationInfos.getSectionCount() < 1 || modificationInfos.getSectionCount() > modificationInfos.getMaximumSectionCount()) {
+            throw new NetworkModificationException(CREATE_SHUNT_COMPENSATOR_ERROR, "Section count should be between 1 and Maximum section count");
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorCreation.java:43]: I think you can add the reel number at the end of the message

File: src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModification.java [hunk 2]
```diff
@@ -46,11 +45,28 @@ public void check(Network network) throws NetworkModificationException {
...
+        int sectionCount = modificationInfos.getSectionCount() != null
+                ? modificationInfos.getSectionCount().getValue()
+                : shuntCompensator.getSectionCount();
+
+        if (modificationInfos.getMaximumSectionCount() != null && modificationInfos.getMaximumSectionCount().getValue() < 1) {
+            throw new NetworkModificationException(MODIFY_SHUNT_COMPENSATOR_ERROR, "Maximum section count should be greater or equal to 1");
+        }
+
+        if (sectionCount < 1 || maximumSectionCount < 1 || sectionCount > maximumSectionCount) {
+            throw new NetworkModificationException(MODIFY_SHUNT_COMPENSATOR_ERROR, "Section count should be between 1 and Maximum section count");
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModification.java:61]: Same

File: src/test/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModificationTest.java [hunk 3]
```diff
@@ -74,6 +74,38 @@ public void testWrongVoltageLevelId() {
...
+        mockMvc.perform(post(getNetworkModificationUri()).content(mapper.writeValueAsString(shuntCompensator)).contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+        assertLogMessage(new NetworkModificationException(MODIFY_SHUNT_COMPENSATOR_ERROR,
+                        String.format("Maximum section count should be greater or equal to 1")).getMessage(),
+                shuntCompensator.getErrorType().name(), reportService);
+    }
+
+    @SneakyThrows
+    @Test
+    public void testWrongSectionCount() {
```
Comment (flomillot) [inline src/test/java/org/gridsuite/modification/server/modifications/ShuntCompensatorModificationTest.java:95]: Maybe it's missing a test where you only modify `sectionCount` ? Your code is supposed to work but it's not tested

File: src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorCreation.java [hunk 4]
```diff
@@ -40,7 +40,7 @@ public void check(Network network) throws NetworkModificationException {
         }
 
         if (modificationInfos.getSectionCount() < 1 || modificationInfos.getSectionCount() > modificationInfos.getMaximumSectionCount()) {
-            throw new NetworkModificationException(CREATE_SHUNT_COMPENSATOR_ERROR, "Section count should be between 1 and Maximum section count");
+            throw new NetworkModificationException(CREATE_SHUNT_COMPENSATOR_ERROR, String.format("Section count should be between 1 and Maximum section count, actual : %d", modificationInfos.getSectionCount()));
```
Comment (flomillot) [inline src/main/java/org/gridsuite/modification/server/modifications/ShuntCompensatorCreation.java:43]: I was thinking of maximumSectionCount x) but both is even better

**Review by flomillot — APPROVED** [review]

**Pull request merged**